x/batch/decorator.go

```diff
@@ -0,0 +1,167 @@
+/*
+Package batch provides batch transaction support
+middleware to support multiple operations in one
+transaction
+*/
+package batch
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/common"
+)
+
+const MaxBatchMessages = 10
+
+//----------------- Decorator ----------------
+//
+// This is just a binding from the functionality into the
+// Application stack, not much business logic here.
+
+// Decorator iterates through batch transaction messages and passes them down the stack
+type Decorator struct {
+}
+
+var _ weave.Decorator = Decorator{}
+
+// NewDecorator returns a batch transaction decorator
+func NewDecorator() Decorator {
+	return Decorator{}
+}
+
+type BatchTx struct {
+	weave.Tx
```

Comment:
q: why do you need to store `weave.Tx`? It would be more straight forward for me to impl the Marshall/ Unmarshal here with Msg.Marshall/Unmarshal or just raising an error rather than returning something unrelated.

Reply:
we don't "store" a weave.Tx, we comply to an interface which is crucial for this whole thing to work.